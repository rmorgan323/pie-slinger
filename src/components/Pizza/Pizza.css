/******  GLOBAL CSS VARIABLES  ******/
/***  COLORS  ***/
/***  FONTS  ***/
* {
  font-size: 20px;
  font-family: "Century Gothic", CenturyGothic, sans-serif;
  outline: none;
  color: #8b5e3c;
  box-sizing: border-box; }

h1 {
  font-family: "tekton-pro", sans-serif;
  line-height: 1.3; }

h2, h3, h4, h5, h6 {
  font-family: "Century Gothic", CenturyGothic, sans-serif;
  line-height: 1.3;
  font-weight: 500; }

h3 {
  color: #8b5e3c;
  font-weight: 900;
  margin-top: 10px;
  line-height: 1.5; }

h1 {
  font-size: 44px; }

h2 {
  font-size: 39px; }

h3 {
  font-size: 34px; }

h4 {
  font-size: 27px; }

h5 {
  font-size: 20px; }

h6 {
  font-size: 12px; }

p {
  font-size: 20px;
  line-height: 1.3;
  font-weight: 300;
  color: #d1be95; }

/***  MIXINS  ***/
.Pizza {
  max-width: 920px;
  margin: auto; }
  .Pizza .get-pizza-button {
    background-color: #be1e2d;
    border-bottom: 6px solid #d43947;
    color: #fff;
    padding: 10px 25px;
    font-weight: 500;
    font-size: 20px;
    display: block;
    margin: 20px auto;
    cursor: pointer;
    transition: all .3s ease-in-out; }
  .Pizza .get-pizza-button:hover {
    background-color: #921723; }
  .Pizza .pizza-container {
    display: flex;
    justify-content: center; }

@media screen and (max-width: 919px) {
  .Pizza .pizza-container {
    flex-direction: column; } }
