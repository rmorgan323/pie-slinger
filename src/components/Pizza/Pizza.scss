@import '../../variables.scss';

.Pizza {
  margin: auto;
  max-width: 920px;

  .get-pizza-button {
    background-color: $red2;
    border-bottom: 6px solid $red1;
    color: #fff;
    cursor: pointer;
    display: block;
    font-size: 20px;
    font-weight: 500;
    margin: 20px auto;
    padding: 10px 25px;
    @include std-transition;
  }

  .get-pizza-button:hover {
    background-color: darken($red2, 10%);
  }

  .pizza-container {
    display: flex;
    justify-content: center;
  }
}

@media screen and (max-width: 919px) {
  .Pizza {
    .pizza-container {
      flex-direction: column;
    }
  }
}