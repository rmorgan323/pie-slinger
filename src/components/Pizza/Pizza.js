import React, { Component } from 'react';
import OrderDetailsLeft from '../OrderDetailsLeft/OrderDetailsLeft';
import OrderDetailsRight from '../OrderDetailsRight/OrderDetailsRight';
import Graphic from '../Graphic/Graphic';
import getPizza from '../../helper/getPizza/getPizza';
import './Pizza.css';

class Pizza extends Component {
  constructor() {
    super();

    this.state = {
    };
  }

  handleGetPizza = async () => {
    const randomPizza = await getPizza();
    this.setState(randomPizza);
  }

  displayGraphic = () => {
    if (this.state.order_id) {
      return (
        <Graphic 
          crust={this.state.crust}
          size={this.state.size}
          toppings={this.state.toppings}
        />
      );
    }
  }

  displayOrderLeft = () => {
    if (this.state.order_id) {
      return (
        <OrderDetailsLeft 
          date={this.state.order_date}
          orderId={this.state.order_id}
          crust={this.state.crust}
          size={this.state.size}
        />
      );
    }
  }

  displayOrderRight = () => {
    if (this.state.order_id) {
      return (
        <OrderDetailsRight 
          toppings={this.state.toppings}
        />
      );
    }
  }

  render () {
    return (
      <div className="Pizza">
        <button 
          className="get-pizza-button"
          onClick={() => this.handleGetPizza()}
        >PIE ME!</button>
        <div className="pizza-container">
          {this.displayOrderLeft()}
          {this.displayGraphic()}
          {this.displayOrderRight()}
        </div>
      </div>
    );
  }
}

export default Pizza;