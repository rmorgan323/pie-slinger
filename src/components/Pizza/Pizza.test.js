import React from 'react';
import { shallow } from 'enzyme';
import Pizza from './Pizza';

describe('Pizza tests', () => {

  it('should match the snapshot', () => {
    const renderedApp = shallow(<Pizza />);
    expect(renderedApp).toMatchSnapshot();
  });
  
});