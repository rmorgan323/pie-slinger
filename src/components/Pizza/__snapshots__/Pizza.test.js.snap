// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pizza tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pizza />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="get-pizza-button"
          onClick={[Function]}
>
          PIE ME!
</button>,
        <div
          className="pizza-container"
>
          
</div>,
      ],
      "className": "Pizza",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "PIE ME!",
          "className": "get-pizza-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "PIE ME!",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            undefined,
            undefined,
          ],
          "className": "pizza-container",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          undefined,
          undefined,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="get-pizza-button"
            onClick={[Function]}
>
            PIE ME!
</button>,
          <div
            className="pizza-container"
>
            
</div>,
        ],
        "className": "Pizza",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "PIE ME!",
            "className": "get-pizza-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "PIE ME!",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              undefined,
              undefined,
            ],
            "className": "pizza-container",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            undefined,
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
